src/huggingface_hub/hf_api.py

```diff
@@ -9525,6 +9529,83 @@ def list_user_following(self, username: str) -> Iterable[User]:
         for followed_user in r.json():
             yield User(**followed_user)
 
+    def auth_check(self, repo_id: str, repo_type: Optional[str] = None, token: Union[bool, str, None] = None) -> bool:
```

Comment:
As discussed by @Wauplin [here](https://github.com/huggingface/huggingface_hub/pull/2497#discussion_r1741954969), using a boolean for auth_check combines different error types into a single 'False' result. Instead, allowing specific exceptions (GatedRepoError, RepositoryNotFoundError) to propagate provides users with more detailed information and enables more precise error handling in their code.